### ISSUE TITLE
Importer arguments

### ISSUE THREAD

**Issue by jareware** [sass/node-sass#917]
I'm working on a reusable (between projects) importer module for `node-sass` and I'd need to pass in arguments. With the [node API](https://github.com/sass/node-sass#examples) this is of course simple, but for the CLI API I see 2 options:

 1. Using a config file that the importer will look for within its pwd
 1. Using custom CLI options, e.g. `node-sass --importer=my-importer.js --myopt-foo=bar`, and reading them back in through `process.argv` in `my-importer.js`

The latter would be nicer IMO (no need for extraneous files), and the CLI doesn't seem to mind options it doesn't recognize, but would there be a cleaner way that I'm just not thinking of here? Or is there a reason why the latter option just isn't an option?

**Comment by am11** [comment]
In `my-importer.js`, isn't it that you have access to `process.argv` containing all the arguments passes to the actual command?

**Comment by jareware** [comment]
That's... exactly what I suggested as the 2nd option. Please read on. ;)

**Comment by am11** [comment]
Yup, exactly what you stated and then this question confused me:

> is there a reason why the latter option just isn't an option?

when we already have access to (global) `process.argv` array in importers.

**Comment by jareware** [comment]
Right, sure! To clarify, I was wondering whether that might be a bad idea, because for example plans on disallowing unknown CLI args (which many apps do, they'll just print the usage info). Or better yet, if someone has an even better idea for how to pass args to an importer that just hasn't occurred to me! :)

**Comment by xzyfer** [comment]
This is something we're going to need to standardise as plugin, functions
and importers gain popularity.

The first thing that cones to mind is adding node sass config into the
package.json. I see more and projects adopting this convention.

**Comment by jareware** [comment]
Actually I think that might be the best option for me as well, for a variety of reasons. Thanks for the input!

Feel free to close this issue, unless you want to keep it as a reminder to standardize this later on.

**Comment by am11** [comment]
`nodeSassConfig` in package.json is used in other build related scenarios as well (https://github.com/sass/node-sass/pull/743). We can standarise the nested members, for instance:

```json
"nodeSassConfig": {
    "importerCliParams": { }
}
```

(the ultimate way would be to propose JSON schema: https://github.com/json-schema/json-schema)

@jareware, feel free to send the PR with a test case and README update. It is a good starting point and these things keep improving over time.

**Comment by jareware** [comment]
Well I'll first try to build it and see if it's *actually* convenient as well, but I'll report back. :)
